polars/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -63,11 +63,24 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
 
     #[cfg(feature = "string_from_radix")]
     // Parse a string number with base _radix_ into a decimal (i32)
-    fn parse_int(&self, radix: Option<u32>) -> Int32Chunked {
+    fn parse_int(&self, radix: u32, strict: bool) -> PolarsResult<Int32Chunked> {
+        use polars_arrow::utils::CustomIterTools;
         let ca = self.as_utf8();
-
-        let f = |s: &str| <i32 as Num>::from_str_radix(s, radix.unwrap_or(2)).unwrap();
-        ca.apply_cast_numeric(f)
+        if strict {
+            let f = |opt_s: Option<&str>| -> PolarsResult<Option<i32>> {
```

Comment:
We could maybe speed this up and take the same branch as `non_strict` and then simply check the null_count afterwards. If the null_count differs from `self`, strict should fail.

Reply:
Uhh that is smart. I will implement that.